### ISSUE TITLE
Confusion about the code_vocab_size in config.py

### ISSUE THREAD

**Issue by jameszhou-gl** [btheodorou99/HALO_Inpatient#20]
Hi authors, nice work! I have one question about the code_vocab_size in config.py. You set code_vocab_size=6841 since there're 6841 unique ICD-9 codes as introduced in the paper. However, I find there're 6444 unique codes in train_ehr_dataset and 3400 unique codes in val_ehr_dataset, and the max values in both train_ehr_dataset and val_ehr_dataset are 6983. 
If follow the default config 6841 for code_vocab_size, there would be an indexError.

So I'm confused where's 6841 from and what's the reasonable value for code_vocab_size in config.py?

https://github.com/btheodorou99/HALO_Inpatient/blob/75b96f5692c1cd55fa88eb3deb31754fc2bc0807/config.py#L11

**Comment by btheodorou-codecertain** [comment]
Hello, I'm not 100% sure of the cause of this (whether it was an outdated config.py file that we uploaded, a different version of MIMIC data since we ran our experiments, or something else). However, to get the code to run as expected, set the code_vocab_size as one greater than the maximum value in the datasets (in this case to 6984). The missing codes should learn to be ignored and never generated. Note the mismatch between the train and val dataset is just a result of having granular and thus rare codes, but it shouldn't mess with how the training works (picking the best version of the model using the validation set will just be based on the overlapping codes). That is the reason we restrict the test set to only training codes when we evaluate, however.

**Comment by jameszhou-gl** [comment]
Thanks for your response.
I use the code with commit id of '75b96f5692c1cd55fa88eb3deb31754fc2bc0807' and mimic-iii-clinical-database-1.4 dataset. Have set code_vocab_size=6984 to run the experiments successfully.

**Comment by jameszhou-gl** [comment]
Yes, I see. I notice you're filtering out any codes that the model has not seen during training in test_model.py:
`test_ehr_dataset = [{'labels': p['labels'], 'visits': [[c for c in v if c in train_c] for v in p['visits']]} for p in test_ehr_dataset]`
but there's no filter for val_ehr_dataset[loc:loc+batch_size].
